assets/tests/components/vehiclePropertiesCard.test.tsx

```diff
@@ -157,6 +163,36 @@ describe("<VehiclePropertiesCard/>", () => {
           screen.getByRole("status", { name: "Current Location" })
         ).toHaveTextContent("Exact location cannot be determined")
       })
+
+      test("when location is loading a new vehicle, should show loading text", () => {
+        const vehicle = vehicleFactory.build()
+        let resolveFn: null | ((value: unknown) => void) = null
+        const locationPromise = new Promise((resolve) => {
+          resolveFn = resolve
+        })
+
+        ;(fetchNearestIntersection as jest.Mock).mockReturnValue(
+          locationPromise
+        )
+
+        render(<VehiclePropertiesCard vehicle={vehicle} onClose={jest.fn()} />)
+
+        expect(
+          screen.getByRole("status", { name: "Current Location" })
+        ).toHaveTextContent(/loading/i)
+
+        const intersection = "intersection ave @ street"
+        resolveFn!(intersection)
+
+        waitFor(
+          () => {
+            expect(
+              screen.getByRole("status", { name: "Current Location" })
+            ).toHaveTextContent(intersection)
+          },
+          { timeout: 2 }
+        )
```

Comment:
@mbta/skate-developers does c606f1dc4f58097e0103033fca33a1b3f91b716c look okay to you? 
Can you think of any other ways to mock promise pending states?

Reply:
Could you mock `useNearestIntersection` instead, since that's what the component invokes directly? That might be simpler.

Reply:
Oh I guess what I forgot was that I'm changing the API of `useNearestIntersection`, so I don't need to mock the promise I just need to mock what I expect it should return in each state 👍🏻 
I was digging myself into a hole trying to figure out how to mock intermediate states heh

Thanks Eddie!

Reply:
[`688eef3` (#1873)](https://github.com/mbta/skate/pull/1873/commits/688eef3397594eae331dbf5ed3a3a3ff15bc8260)
and
[`b6fad54` (#1873)](https://github.com/mbta/skate/pull/1873/commits/b6fad54cd7da76701ad163626f68a02805fe9da6)